### ISSUE TITLE
Choropleth map plot does not work if all the state values are the same (R library)

### ISSUE THREAD

**Issue by benjaminysmith** [cmu-delphi/covidcast#134]
When trying to plot data that has the same value for every state I get: "Error: Breaks and labels are different lengths". Changing one value slightly allows the plot. I suspect this is because of the way that the legend breaks vs labels are calculated.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1198968030437421/1198970434114006) by [Unito](https://www.unito.io/learn-more)

**Comment by ryantibs** [comment]
@benjaminysmith Can you produce example code?  
@JedGrabman maybe you want to take a quick look?

**Comment by benjaminysmith** [comment]
Example:

```
library(covidcast)

example_df = data.frame("geo_value"= tolower(c(state.abb)))

example_df$time_value = "2020-04-15"
example_df$issue = "2020-04-15"
example_df$value = 10
attributes(example_df)$geo_type = "state"
class(example_df) = c("covidcast_signal", "data.frame")

plot(example_df)
```

Fails with:

`Error: Breaks and labels are different lengths`

Adding `range=c(0,10)` to plot command is workaround.

**Comment by JedGrabman** [comment]
```
In addition: Warning message:
Metadata for signal mean and standard deviation not available; defaulting to observed mean and standard deviation to set plot range.
```

Issue looks like it probably starts [here](https://github.com/cmu-delphi/covidcast/blob/5ca16a9e111b82f23c399549a8e7d5f4c910bcfd/R-packages/covidcast/R/covidcast.R#L414)


`stdev_value <- sd(x$value)`
...
`range <- c(mean_value - 3 * stdev_value, mean_value + 3 * stdev_value)`

If we just check if stdev is 0 or set it to atleast some small positive value, it should take care of it.

**Comment by JedGrabman** [comment]
Adding to myself for now so it doesn't fall through the cracks, but treating as low-priority

**Comment by statsmaths** [comment]
The older counterexample doesn't run for a new reason now (I think due to some naming changes in the package). Here's a way to reproduce on r-pkg-devel:

```{r}
library(covidcast)
library(tibble)

example_df <- structure(
  tibble(
    geo_value = tolower(state.abb),
    time_value = "2020-04-15",
    issue = "2020-04-15",
    data_source = "example",
    signal = "example",
    value = 10
  ),
  metadata = list(geo_type = "state"),
  class = c("covidcast_signal", "data.frame")
)

plot(example_df)
```

**Comment by statsmaths** [comment]
This should be fixed in #504, which converts a zero standard deviation to 10% of the mean or (if the mean is also zero) to 0.001.

**Comment by statsmaths** [comment]
Fixed in #504.
